### PR TITLE
feat(casl): implement casl

### PR DESCRIPTION
**Pull request by r-hannuschka**
- [x] e2e tests

### CODE REVIEW

File: apps/server/src/app/auth/utils/jwt-auth.strategy.ts [hunk 1]
```diff
@@ -14,8 +15,13 @@ export class JwtStrategy extends PassportStrategy(Strategy) {
     });
   }
 
-  async validate(payload: any) {
-    return true;
-    return { userId: payload.sub, username: payload.username };
+  async validate(payload: User & { exp: number, iat: number }): Promise<User> {
+    // this validates the user by default should we call the auth service to do that ?
+    const { exp, iat, ...user } = payload
+    if (iat < exp) {
```
Comment (konne) [inline apps/server/src/app/auth/utils/jwt-auth.strategy.ts:21]: please use proper jwt check:
1. check if the signature valid
2. time is >= nbf (not before timestamp)
3. time is <= exp (expire timestamp)

iat is the issued at timestamp

**Review by konne — CHANGES_REQUESTED** [review]

File: apps/server/src/app/roles/constants/actions.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export enum Actions {
+  Create = 'create',
+  Read = 'read',
+  Update = 'update',
+  Destroy = 'destroy',
```
Comment (konne) [inline apps/server/src/app/roles/constants/actions.ts:5]: please rename to delete

File: apps/server/src/app/roles/providers/ability.factory.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    // should be role instead of userrole this is a bit duplicated
+    switch (user.role) {
+      case UserRoles.Admin:
+        can(Actions.Manage, 'all')
+        break
+
+      // can not update users
+      case UserRoles.Write:
+        can(Actions.Update, 'all')
```
Comment (konne) [inline apps/server/src/app/roles/providers/ability.factory.ts:23]: not all, Write has only rights to write into the
Set Parameter Entpoints in connection, but for example not 
for changing the connection params,...

File: package.json [hunk 4]
```diff
@@ -28,6 +28,7 @@
     "@angular/platform-browser": "13.2.4",
     "@angular/platform-browser-dynamic": "13.2.4",
     "@angular/router": "13.2.4",
+    "@casl/ability": "^5.4.3",
```
Comment (konne) [inline package.json:31]: pin dependency

**Review by konne — CHANGES_REQUESTED** [review]

File: apps/server/src/app/roles/constants/actions.ts (hunk 2 above)
Comment (r-hannuschka) [reply apps/server/src/app/roles/constants/actions.ts:5]: done

File: package.json (hunk 4 above)
Comment (r-hannuschka) [reply package.json:33]: done

**Review by konne — APPROVED** [review]

**Pull request merged**